### PR TITLE
feat: Add JSON schema to `elasticsearch` destination

### PR DESCRIPTION
**Pull request by disq**
Closes https://github.com/cloudquery/cloudquery/issues/16448

### CODE REVIEW

File: plugins/destination/elasticsearch/client/spec.go [hunk 1]
```diff
@@ -1,31 +1,54 @@
...
+	// Number of bytes to batch together per request.
+	BatchSizeBytes int `json:"batch_size_bytes" jsonschema:"minimum=1,default=5242880"`
 }
 
+//go:embed schema.json
+var JSONSchema string
+
 func (s *Spec) SetDefaults() {
 	if len(s.Addresses) == 0 {
 		s.Addresses = []string{"http://localhost:9200"}
```
Comment (disq) [inline plugins/destination/elasticsearch/client/spec.go:54]: This is very basic and Validate() is empty... doesn't make a good json schema :)

File: plugins/destination/elasticsearch/client/schema.json [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    "Spec": {
+      "properties": {
+        "addresses": {
+          "oneOf": [
+            {
+              "items": {
+                "type": "string"
+              },
+              "type": "array",
+              "description": "A list of Elasticsearch nodes to use. (Default: [\"http://localhost:9200\"])"
```
Comment (candiduslynx) [inline plugins/destination/elasticsearch/client/schema.json:15]: you can actually use `default` here

File: plugins/destination/elasticsearch/client/schema.json [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+          "type": "string",
+          "description": "Username for HTTP Basic Authentication."
+        },
+        "password": {
+          "type": "string",
+          "description": "Password for HTTP Basic Authentication."
+        },
+        "cloud_id": {
+          "type": "string",
+          "description": "Endpoint for the Elastic Service (https://elastic.co/cloud)."
```
Comment (candiduslynx) [inline plugins/destination/elasticsearch/client/schema.json:32]: wrong desc?

File: plugins/destination/elasticsearch/client/spec.go [hunk 4]
```diff
@@ -1,31 +1,54 @@
...
 	defaultBatchSize      = 1000
 	defaultBatchSizeBytes = 5 * 1024 * 1024
 )
 
 type Spec struct {
-	Addresses []string `json:"addresses"` // A list of Elasticsearch nodes to use.
-	Username  string   `json:"username"`  // Username for HTTP Basic Authentication.
-	Password  string   `json:"password"`  // Password for HTTP Basic Authentication.
+	// A list of Elasticsearch nodes to use. (Default: ["http://localhost:9200"])
+	Addresses []string `json:"addresses"`
```
Comment (candiduslynx) [inline plugins/destination/elasticsearch/client/spec.go:15]:
```suggestion
	Addresses []string `json:"addresses" jsonschema:"default=http://localhost:9200"`
```

File: plugins/destination/elasticsearch/client/spec.go [hunk 5]
```diff
@@ -1,31 +1,54 @@
...
+
+	// Username for HTTP Basic Authentication.
+	Username string `json:"username"`
+
+	// Password for HTTP Basic Authentication.
+	Password string `json:"password"`
+
+	// Endpoint for the Elastic Service (https://elastic.co/cloud).
```
Comment (candiduslynx) [inline plugins/destination/elasticsearch/client/spec.go:23]: is it a proper desc?

**Review by candiduslynx — CHANGES_REQUESTED** [review]

File: plugins/destination/elasticsearch/client/schema.json (hunk 3 above)
Comment (disq) [reply plugins/destination/elasticsearch/client/schema.json:32]: No it's correct.

File: plugins/destination/elasticsearch/client/spec.go (hunk 5 above)
Comment (disq) [reply plugins/destination/elasticsearch/client/spec.go:23]: That's what the doc says.

File: plugins/destination/elasticsearch/client/spec.go [hunk 6]
```diff
@@ -1,31 +1,54 @@
...
 const (
 	defaultBatchSize      = 1000
 	defaultBatchSizeBytes = 5 * 1024 * 1024
 )
 
 type Spec struct {
-	Addresses []string `json:"addresses"` // A list of Elasticsearch nodes to use.
-	Username  string   `json:"username"`  // Username for HTTP Basic Authentication.
-	Password  string   `json:"password"`  // Password for HTTP Basic Authentication.
+	// A list of Elasticsearch nodes to use. (Default: ["http://localhost:9200"])
```
Comment (candiduslynx) [inline plugins/destination/elasticsearch/client/spec.go:14]: No need for default repetition here.
```suggestion
	// A list of Elasticsearch nodes to use.
```

**Review by candiduslynx — APPROVED** [review]
only a small nit

**Review by candiduslynx — APPROVED** [review]

**Pull request merged**